server/src/main/java/io/unitycatalog/server/handler/CoordinatedCommitsHandler.java

```diff
@@ -0,0 +1,140 @@
+    //    assert tableInfo.getTableType() == TableType.MANAGED;
+    assert tableInfo.getDataSourceFormat() == DataSourceFormat.DELTA;
+
+    // TODO: Add other assertions like verifying the table path (tableInfo.getStorageLocation)
+  }
+
+  public static void validateOnboardingCommit(Commit commit) {
+    assert commit.getCommitInfo() != null;
+    // Onboarding commit cannot be a disown commit
+    assert !commit.getCommitInfo().getIsDisownCommit();
+  }
+
+  public static void handleFirstCommit(Commit commit) {
+    validateOnboardingCommit(commit);
+    COMMIT_REPOSITORY.saveCommit(commit);
+  }
+
+  public static void handleBackfillOnlyCommit(
+      String tableId, Long latestBackfilledVersion, CommitDAO firstCommit, CommitDAO lastCommit) {
+    // Nothing to delete, return
+    if (latestBackfilledVersion < firstCommit.getCommitVersion()) {
+      return;
+    }
+    // We only need to delete when there is more than one commit. We always keep the last commit.
+    if (firstCommit.getCommitVersion() < lastCommit.getCommitVersion()) {
+      COMMIT_REPOSITORY.backfillCommits(
+          UUID.fromString(tableId),
+          Math.min(latestBackfilledVersion, lastCommit.getCommitVersion() - 1),
+          firstCommit,
+          lastCommit.getCommitVersion());
+    }
+    if (latestBackfilledVersion == lastCommit.getCommitVersion()) {
+      // Mark the last commit as the latest backfilled version
+      COMMIT_REPOSITORY.markCommitAsLatestBackfilled(
+          UUID.fromString(tableId), lastCommit.getCommitVersion());
+    }
+
+    // TODO: Should we also retain the disown commit if it's backfilled?
+  }
+
+  public static void handleReboardCommit() {
+    throw new RuntimeException("Reboarding of tables not allowed!");
+  }
+
+  public static void handleNormalCommit(
+      Commit commit, CommitDAO firstCommit, CommitDAO lastCommit) {
+    if (commit.getCommitInfo().getVersion() <= lastCommit.getCommitVersion()) {
+      throw new RuntimeException(
+          "Commit version should be greater than the last commit version = "
+              + lastCommit.getCommitVersion());
+    }
+    if (commit.getCommitInfo().getVersion() > lastCommit.getCommitVersion() + 1) {
+      throw new RuntimeException(
+          "Commit version should be the next version after the last commit version = "
+              + lastCommit.getCommitVersion());
+    }
+    if (commit.getLatestBackfilledVersion() != null
+        && commit.getLatestBackfilledVersion() > lastCommit.getCommitVersion()) {
+      throw new RuntimeException(
+          "Latest backfilled version cannot be greater than the last commit version = "
+              + lastCommit.getCommitVersion());
+    }
+    long latestBackfilledVersion = getLatestBackfilledVersion(commit, firstCommit, lastCommit);
+    if (commit.getCommitInfo().getVersion() - latestBackfilledVersion > MAX_NUM_COMMITS_PER_TABLE) {
+      throw new RuntimeException(
```

Comment:
Let's properly define this exception somewhere instead of throwing a generic RuntimeException.